### ISSUE TITLE
[LightwaveRF] energy monitor does not connect and eventually crashes OpenHab 

### ISSUE THREAD

**Issue by curtash** [openhab/openhab1-addons#5226]
## Expected Behavior

I have the 2nd version of the lightwaveRF hub (the one without a screen) and the updated LightwaveRF energy monitor. 

It works fine in the official App along with all my other LW products.

All switches and plugs work in OpenHab but the energy monitor keeps failing.

## Current Behavior

As soon as I plug in the monitor and even without adding anything in the "items" or "sitemap" my openhab.log gets swamped with error messages and after a couple of days my Pi crashes.

**The Error**
`[ERROR] [twaverf.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":72886,"mac":"20:30:18","time":1497600860,"pkt":"868R","fn":"meterData","prod":"pwrMtr","serial":"4193FE","type":"energy","cUse":2766,"todUse":6116}
`

**My sitemap:**
`Text item=Energy_Current_Usage`

**My items:**
`Number Energy_Current_Usage "Mains usage: Current [%.1f W]" 	
{ lightwaverf="serial=4193FE,type=ENERGY_CURRENT_USAGE" }`

## Your Environment

* org.openhab.binding.lightwaverf-1.10.0-SNAPSHOT.jar
* openhab2
* Pi3 

## Note

1. Others with this issue can be found here https://community.openhab.org/t/lightwave-rf-energy-monitor/18960

1. Happy to help test this in any way I can, just contact me on the link above or here

**Comment by 9037568** [comment]
Can you provide a debug log?

**Comment by curtash** [comment]
Sure is that "openhab.log". Sorry I'm pretty new to this.

**Comment by 9037568** [comment]
Yes, but you need to set the log level for the binding to DEBUG. See [the docs](http://docs.openhab.org/administration/logging.html#defining-what-to-log) here.

````
log:set DEBUG org.openhab.binding.lightwaverf
````

**Comment by curtash** [comment]
- Found the file here: 
`sudo vi /var/lib/openhab2/etc/org.ops4j.pax.logging.cfg`

- Added this to the top of the file:
`log:set DEBUG org.openhab.binding.lightwaverf`

- deleted my "openhab.log"
- shutdown Pi
- switched LW energy monitor on
- checked energy monitor working in LW app
- booted Pi
- waited 10min and attached the logs here

Thought I would explain exactly what I did just incase I did something incorrectly.  My logs don't seem any different but that is possibly cause I don"t know what to look for.

[openhab.txt](https://github.com/openhab/openhab1-addons/files/1097767/openhab.txt)

**Comment by 9037568** [comment]
I don't think you can stuff a log:set command into a log4j config file like that.  You should either issue it at the prompt in the OH2 console, or use this in the pax.logging.cfg file:

NOTE: I haven't tested this log config method, but it should work in theory...
````
# openHAB specific logger configuration
...
log4j.logger.org.openhab = INFO
...
log4j.logger.org.openhab.binding.lightwaverf = DEBUG
````

Once the log level is correctly configured, you should see some messages in the log file tagged as `[DEBUG]` and possibly having content similar to:

````
internalReceiveCommand(...) is called!
internalReceiveUpdate(...) is called!
Not sending command...
Publishing Update ...
Not publishing message...
````

**Comment by curtash** [comment]
Cheers, that seemed to do the trick. New log attached.

[openhab.txt](https://github.com/openhab/openhab1-addons/files/1099522/openhab.txt)

**Comment by 9037568** [comment]
A lot of digging through code appears to indicate this is failing because the binding expects there to be a "signal" attribute in the message. The message you have does not have any such attribute.

Here's a message that does:
````
*!{"trans":215955,"mac":"03:41:C4","time":1435620183,"prod":"pwrMtr","serial":"9470FE","signal":79,"type":"energy","cUse":271,"maxUse":2812,"todUse":8414,"yesUse":8377}'
````

And here's the one you're getting:
````
*!{"trans":72886,"mac":"20:30:18","time":1497600860,"pkt":"868R","fn":"meterData","prod":"pwrMtr","serial":"4193FE","type":"energy","cUse":2766,"todUse":6116}
````

**Comment by curtash** [comment]
Thanks Chris, your time is really appreciated.

Strange, so what is the solution? Something I can do my end or something that needs to be done to the binding?

**Comment by 9037568** [comment]
Well, ideally, the whole binding would be rewritten to do proper data validation.

But before we try biting all that off, let's see if a smaller patch can resolve the issue with the power monitor without breaking everything else.

I'll build a test jar for you before the end of the weekend.

**Comment by 9037568** [comment]
Try [this jar](https://drive.google.com/open?id=0B8vYheWfObP9X25kNk1wd2l4cWs).

**Comment by curtash** [comment]
Updated log file after adding new jar
[openhab.txt](https://github.com/openhab/openhab1-addons/files/1100397/openhab.txt)

Do you want my to add an item or is that not required yet?

**Comment by 9037568** [comment]
Sure, go ahead and set up as much lightwave stuff as you would want to use, and see if it works properly.

**Comment by curtash** [comment]
Everything has been added (about 20 items). My energy monitor is now reporting values 👍 . 

I'll leave it 24 hours and see if it crashes OpenHab.

Currently this is what is appearing in my logs about every 10 seconds, if it helps.

`2017-06-25 18:41:22.780 [DEBUG] [f.internal.LightwaveRFReceiverThread] - Message received from: /192.168.0.4 message:*!{"trans":11574,"mac":"20:30:18","time":1498416081,"pkt":"868R","fn":"meterData","prod":"pwrMtr","serial":"4193FE","type":"energy","cUse":1610,"todUse":8582}
2017-06-25 18:41:22.786 [WARN ] [mand.LightwaveRfEnergyMonitorMessage] - Failed to get a signal attribute from the message '*!{"trans":11574,"mac":"20:30:18","time":1498416081,"pkt":"868R","fn":"meterData","prod":"pwrMtr","serial":"4193FE","type":"energy","cUse":1610,"todUse":8582}'
2017-06-25 18:41:22.789 [WARN ] [mand.LightwaveRfEnergyMonitorMessage] - Failed to get a maxUse attribute from the message '*!{"trans":11574,"mac":"20:30:18","time":1498416081,"pkt":"868R","fn":"meterData","prod":"pwrMtr","serial":"4193FE","type":"energy","cUse":1610,"todUse":8582}'
2017-06-25 18:41:22.792 [WARN ] [mand.LightwaveRfEnergyMonitorMessage] - Failed to get a yesUse attribute from the message '*!{"trans":11574,"mac":"20:30:18","time":1498416081,"pkt":"868R","fn":"meterData","prod":"pwrMtr","serial":"4193FE","type":"energy","cUse":1610,"todUse":8582}'
2017-06-25 18:41:22.794 [DEBUG] [htwaverf.internal.LightwaveRfBinding] - Publishing Update org.openhab.binding.lightwaverf.internal.command.LightwaveRfEnergyMonitorMessage@1d22229 to [Energy_Yesterday_Usage, Energy_UpdateTime, Energy_Current_Usage, Energy_Today_Usage, Energy_Max_Usage]
2017-06-25 18:41:22.799 [INFO ] [htwaverf.internal.LightwaveRfBinding] - Update from LightwaveRf ItemName[Energy_Yesterday_Usage], State[0]
2017-06-25 18:41:22.836 [INFO ] [htwaverf.internal.LightwaveRfBinding] - Update from LightwaveRf ItemName[Energy_UpdateTime], State[2017-06-25T18:41:22]
2017-06-25 18:41:22.844 [INFO ] [htwaverf.internal.LightwaveRfBinding] - Update from LightwaveRf ItemName[Energy_Current_Usage], State[1610]
2017-06-25 18:41:22.847 [INFO ] [htwaverf.internal.LightwaveRfBinding] - Update from LightwaveRf ItemName[Energy_Today_Usage], State[8582]
2017-06-25 18:41:22.852 [INFO ] [htwaverf.internal.LightwaveRfBinding] - Update from LightwaveRf ItemName[Energy_Max_Usage], State[0]`

**Comment by 9037568** [comment]
Cool. I knocked those WARNs down to DEBUGs and updated the jar file.

Source code change can be seen in [my fork](https://github.com/9037568/openhab/commit/53f62ef864d0b185771477a491c936f55bad17db).

**Comment by curtash** [comment]
Hi Chris, It's been on 2 days and OpenHab has not crashed and my monitor is reporting values. 

Am I to understand from your previous message that you have updated the binding again to stop filling the log up with errors? If so I'm not sure of the process to get them from your fork to my binding. I thought I may be able to copy the code in your fork into my binding but the file has .class and not .java so I assume I can't do it the way I thought. 

Please can you point me in the right direction as my logs are getting hammered. I am aware in need to switch debug mode off too.

[openhab.txt](https://github.com/openhab/openhab1-addons/files/1107772/openhab.txt)

I really appreciate your help with this thanks again.

On a final note I was wondering what controls how often my energy monitor reports values. Am I correct in assuming that it is the device the controls how often it submits a message. Was just wondering if I could have set it to report, say, every minute rather that every 10 seconds.

**Comment by 9037568** [comment]
I updated the jar file at the link I provided earlier so that the log messages are now DEBUG level instead of WARN.

So if you redownload and reinstall the jar and set your OH log levels back to normal, your log should stop seeing the messages.

As to what controls how often your energy monitor reports values, I would assume that's a configuration thing in the energy monitor and not related to OH. I don't see any evidence OH is polling them.
